homeassistant/components/flexpool/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow for Flexpool integration."""
+import logging
+import re
+
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from ...const import CONF_ADDRESS
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ADDRESS): str,
+        vol.Optional("workers"): bool,
+        vol.Optional("pool"): bool,
```

Comment:
Workers and pool are controlling if certain entities are created or not.
This is should not be part of the config flow. In the config we only ask the bare minimum that is needed to set up the integration.

Entity visibility is controlled via the entity registry and a default feature of Home Assistant (no need to control them here).

```suggestion
```

Reply:
The problem with that is that if a user has, say 100 workers, that would clutter their instance with x4 entities quite a lot. I agree on the pool part, however workers should be controllable

Reply:
If you think that some entities are not needed, they still should be created and flagged disabled by default. As said, the config flow should contain the bare minimum, and these are not required questions to get something working.